src/logging.cc

```diff
@@ -1327,8 +1327,7 @@ void LogCleaner::Run(bool base_filename_selected,
   if (!base_filename_selected) {
     dirs = GetLoggingDirectories();
   } else {
-    size_t pos = base_filename.find_last_of(possible_dir_delim, 0,
-                                            sizeof(possible_dir_delim));
+    size_t pos = base_filename.find_last_of(possible_dir_delim);
```

Comment:
I do not believe this to be a valid fix because `possible_dir_delim` is not null-terminated. Thus you cannot leave the last `count` parameter out. According to [cppreference](https://en.cppreference.com/w/cpp/string/basic_string/find_last_of), this parameter does not specify how many characters are searched for in the source string but rather in the character string identifying characters to search for.

Reply:
Ah, sorry, I confused the second parameter `pos` of `std::string::find_last_of` with that of `std::string::find_first_of`.

Only the second argument must be changed (from 0 to string::npos)